lib/task/import/csvAccessionImportTask.class.php

```diff
@@ -406,65 +406,51 @@ public function execute($arguments = [], $options = [])
         });
 
         $import->addColumnHandler('resourceType', function ($self, $data) {
-            if ($data) {
-                $data = trim($data);
-                $resourceTypeId = self::arraySearchCaseInsensitive($data, $self->status['resourceTypes'][$self->columnValue('culture')]);
-
-                if (false === $resourceTypeId) {
-                    echo "\nTerm {$data} not found in resource type taxonomy, creating it...\n";
-                    $newTerm = QubitFlatfileImport::createTerm(QubitTaxonomy::ACCESSION_RESOURCE_TYPE_ID, $data, $self->columnValue('culture'));
-                    $self->status['resourceTypes'] = self::refreshTaxonomyTerms(QubitTaxonomy::ACCESSION_RESOURCE_TYPE_ID);
-                }
-
-                $this->setObjectPropertyToTermIdLookedUpFromTermNameArray(
-                    $self,
-                    'resourceTypeId',
+            if ($data && isset($self->object) && $self->object instanceof QubitAccession) {
+                $self->object->resourceTypeId = $self->createOrFetchTermIdFromName(
```

Comment:
Ooh, I like that you moved the "create term" logic into a method to avoid repetition. :)